src/theme/src/hooks/useCheckboxAppearance.js

```diff
@@ -1,140 +1,35 @@
- * @return {Object} the appearance of the checkbox.
- */
-const getCheckboxStyles = theme => {
-  const {
-    tokens: { primary, colors },
-    checkbox
-  } = theme
-
-  const {
-    base: baseStyles = {},
-    disabled: disabledStyles = {},
-    hover: hoverStyles = {},
-    focus: focusStyles = {},
-    active: activeStyles = {},
-    checked: checkedStyles = {},
-    checkedHover: checkedHoverStyles = {},
-    checkedDisabled: checkedDisabledStyles = {},
-    checkedActive: checkedActiveStyles = {}
-  } = checkbox || {}
-
-  return {
-    ...hiddenCheckboxStyle,
-    '& + div > svg': { display: 'none' },
-    [defaultState]: {
-      ...baseStyle,
-      color: 'white',
-      backgroundColor: 'white',
-      boxShadow: `inset 0 0 0 1px ${colors.gray400}`,
-      ...baseStyles
-    },
-    [disabledState]: {
-      cursor: 'not-allowed',
-      backgroundColor: colors.gray100,
-      boxShadow: `inset 0 0 0 1px ${colors.gray100}`,
-      ...disabledStyles
-    },
-    [hoverState]: {
-      boxShadow: `inset 0 0 0 1px ${colors.gray600}`,
-      ...hoverStyles
-    },
-    [focusState]: {
-      boxShadow: `0 0 0 2px ${colors.blue100}`,
-      ...focusStyles
-    },
-    [activeState]: {
-      backgroundColor: colors.gray100,
-      boxShadow: `inset 0 0 0 1px ${colors.gray500}`,
-      ...activeStyles
-    },
-    [checkedState]: {
-      ...checkedStyle,
-      color: 'white',
-      boxShadow: `inset 0 0 0 -1px ${primary.active}`,
-      backgroundColor: primary.base,
-      ...checkedStyles
-    },
-    [checkedHoverState]: {
-      ...checkedStyle,
-      color: 'white',
-      backgroundColor: primary.hover,
-      boxShadow: `inset 0 0 0 1px ${primary.hover}`,
-      ...checkedHoverStyles
-    },
-    [checkedDisabledState]: {
-      ...checkedStyle,
-      color: colors.gray600,
-      backgroundColor: colors.gray100,
-      ...checkedDisabledStyles
-    },
-    [checkedActiveState]: {
-      ...checkedStyle,
-      color: 'white',
-      boxShadow: `inset 0 0 0 -1px ${primary.active}`,
-      backgroundColor: primary.active,
-      ...checkedActiveStyles
-    }
-  }
+const pseudoSelectors = {
+  _display: '& + div',
+  _tickSvg: '& + div > svg',
+  _disabled: '&:disabled + div',
+  _hover: '&:not(:disabled):hover + div',
+  _focus: '&:not(:disabled):focus + div',
+  _active: '&:not(:disabled):active + div',
+  _checked: '&:checked + div, &[type=checkbox]:indeterminate + div',
+  _checkedHover: '&:not([disabled]):checked:hover + div, &[type=checkbox]:not([disabled]):indeterminate:hover + div',
+  _checkedActive: '&:not([disabled]):checked:active + div, &[type=checkbox]:not([disabled]):indeterminate:active + div',
+  _checkedDisabled: '&[disabled]:checked + div, &[type=checkbox][disabled]:indeterminate + div'
 }
 
 /**
  * Get the className of a `Checkbox`.
  * @param {string} appearance
  * @return {string} the appearance class name.
  */
-function useCheckboxAppearance() {
+function useCheckboxAppearance(modifiers) {
```

Comment:
Sweet - does `size` make sense as a modifier for input components?

Reply:
IMO yes! We can bring that as a follow up PR if that's too many changes though :)